src/main.rs

```diff
@@ -1085,6 +1137,49 @@ fn file_is_overwritable(path: &Path) -> bool {
 //     add_diagnostic("Successfully parsed %d / %d pages" % (successful_count, total_count), BRIEF_VERBOSE)
 //     flush_diagnostics(sys.stderr)
 
+#[derive(Copy, Clone, Debug)]
+struct Progress(pub bool);
+
+// TODO Arg/output types?
+fn parse_and_output_man_pages(
+    _paths: impl Iterator<Item = PathBuf>,
+    _output_directory: PathBuf,
+    Progress(_show_progress): Progress,
+) {
+    unimplemented!();
+}
+
+fn parsers_to_try() -> Vec<Box<dyn ManParser>> {
+    vec![
+        Box::new(Type1),
+        Box::new(Type2),
+        Box::new(Type3),
+        Box::new(Type4),
+        Box::new(TypeDarwin),
+        Box::new(TypeDeroff),
+    ]
+}
+
+use std::io::{Read, Write};
```

Comment:
Why was this moved here?

Reply:
No great reason. Needed `Write` for the function below and was resolving a warning for it being `use`d twice.

I've amended main.rs to move all the root-level `use` statements to the top of the file.